.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,57 @@
+
++ for the curious, the Wikipedia article on [isolation in database systems](https://en.wikipedia.org/wiki/Isolation_(database_systems)) (but this isn't something you need to know anything about since the `readOnly`/`readWrite` abstractions exist)
+
+##### Database
+The services layer defines all the rules of our portal statelessly&mdash;our ACM UCSD data could be replaced by another chapter's data and the portal would still function the same&mdash;, so it's our data that makes up the state of our application. As such, it's important to maintain a healthy application state and protect against data loss and general weirdness; we keep database backups as often as possible, thoroughly test any changes we make to our data, and use a relational database (PostgreSQL) to ensure our data conforms to a strict and unchanging schema (among the other usual reasons).
+
+Modifying the schema requires not only direct edits to the data models (found in the `models` folder) but also schema migrations (found in the `migrations` folder). Schema migrations are procedures for changing the schema from an older version (such as the schema currently running in production) to your modified version. To write a schema migration, take a look at the [relevant TypeORM documentation](https://github.com/typeorm/typeorm/blob/master/docs/migrations.md) and [existing migrations](https://github.com/acmucsd/membership-portal/tree/master/migrations). Everything's already configured via [`ormconfig.ts`](https://github.com/acmucsd/membership-portal/blob/master/ormconfig.ts) so running `npm run db:migrate` will execute your migration(s). Finally, we have our data repositories, which contain predefined data queries that abstract away database interactions such as complex SQL queries and can be easily called by services, in the `repositories` folder. All data in use by the application should be immutable, and the services should not interact with the database except through repository methods. This means preferring `userRepository.update(user, changes)` to `user.field = value; user.save()`.
```

Comment:
Maybe bold the first sentence? I think generally there should be more bolding since I'm not sure how many people would read every single paragraph. Important things like validation, migrations, transactions, and others I think are worth bolding for importance

Reply:
There's no bolding in the project section because first-time contributors should read all of it.